gossip/src/duplicate_shred.rs

```diff
@@ -176,7 +199,12 @@ where
         return Err(Error::InvalidDuplicateShreds);
     }
     let other_shred = Shred::new_from_serialized_shred(other_payload)?;
-    check_shreds(leader_schedule, &shred, &other_shred)?;
+    check_shreds(
+        leader_schedule,
+        &shred,
+        &other_shred,
+        /* allow_chained_duplicate_proofs */ true,
```

Comment:
we don't need to feature gate this because it is already gated in window service.

Reply:
can you please include this comment in the code.

Reply:
also this is pretty risky assumption that the caller will not invoke this code if the feature is not activated.
Is it possible to expose this as an argument to this function and let a caller higher in the stack set it to true.